cli/module_registry.go

```diff
@@ -0,0 +1,131 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+// These types mirror app's internal representations of modules
+
+// ModuleVisibility determines whether modules are public or private.
+type ModuleVisibility string
+
+// Permissions enumeration.
+const (
+	ModuleVisibilityPrivate ModuleVisibility = "private"
+	ModuleVisibilityPublic  ModuleVisibility = "public"
+)
+
+// ModuleComponent represents an api - model pair.
+type ModuleComponent struct {
+	API   string `json:"api"`
+	Model string `json:"model"`
+}
+
+// ModuleManifest is used to create & parse manifest.json.
+type ModuleManifest struct {
+	Name        string            `json:"name"`
+	Visibility  ModuleVisibility  `json:"visibility"`
+	URL         string            `json:"url"`
+	Description string            `json:"description"`
+	Models      []ModuleComponent `json:"models"`
+	Entrypoint  string            `json:"entrypoint"`
+}
+
+const (
+	defaultManifestFilename = "meta.json"
+)
+
+// CreateModuleCommand runs the command to create a module
+// This includes both a gRPC call to register the module on app.viam.com and creating the manifest file.
+func CreateModuleCommand(c *cli.Context) error {
+	moduleNameArg := c.String("name")
+	orgIDArg := c.String("org_id")
+	publicNamespaceArg := c.String("public_namespace")
+
+	client, err := NewAppClient(c)
+	if err != nil {
+		return err
+	}
+	publicNamespace, err := resolvePublicNamespace(client, orgIDArg, publicNamespaceArg)
+	if err != nil {
+		return err
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "failed to find the current directory")
+	}
+	// Check to make sure the user doesn't accidentally overwrite a module manifest
+	manifestFilepath := filepath.Join(cwd, defaultManifestFilename)
+	if _, err := os.Stat(manifestFilepath); err == nil {
+		return errors.Errorf("Another module's %v already exists in the current directory. Delete it and try again", defaultManifestFilename)
+	}
+
+	response, err := client.CreateModule(moduleNameArg, publicNamespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to register the module on app.viam.com")
+	}
+	if response.GetModuleId() == "" {
+		return errors.New("empty module id returned. Please report this")
+	}
+
+	fmt.Fprintf(c.App.Writer, "Successfully created '%s'.\n", response.GetModuleId())
+	if response.GetUrl() != "" {
+		fmt.Fprintf(c.App.Writer, "You can view it here: %s \n", response.GetUrl())
+	}
+
+	emptyManifest := ModuleManifest{
+		Name:       response.GetModuleId(),
+		Visibility: ModuleVisibilityPrivate,
+		// This is done so that the json has an empty example
```

Comment:
I think lets add all fields with an empty string. Its easier to fill out then. I put an example in https://docs.google.com/document/d/1fKc237hWceeez0RDfJk8yrS39P67az5ffN_ToXgMGWs/edit#heading=h.kbl4j7zxrdo

Reply:
This already does that. I don't have any omitempty so all of the fields end up having an empty string. I have an example up above. 

Not sure if I should add omitempty though.

Reply:
Ah ok, by bad. Makes total sense now. Was wondering why the code didn't have them but your PR overview did :)

Reply:
Ok to resolve?